packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -220,6 +223,45 @@ export async function executeWithContext<
       createStepGraphObjectDataUploader,
     } = options;
 
+    // Load data for ignored step dependencies.
+    if (options.useStorageForIgnoredStepDependencies) {
```

Comment:
Oops, forgot to add this comment to the review - it would be really nice to abstract this behavior behind a function.

```ts
if (options.useStorageForIgnoredStepDependencies) {
  await populateJobStateFromCache({ ... });
}
```